km/km_filesys.c

```diff
@@ -2537,7 +2537,13 @@ static inline int km_fs_recover_socket(km_nt_socket_t* nt_sock, struct sockaddr*
       return -1;
    }
    km_fs_recover_fd(nt_sock->fd, host_fd, 0, km_get_nonfile_name(host_fd), -1, nt_sock->how);
-
+   if (nt_sock->state == KM_NT_SKSTATE_ERROR) {
+      file->error = -ECONNRESET;
+      if (file->name != NULL) {
```

Comment:
free() tolerates being passed a null pointer.  The man page even says so.

PS: this usually causes endless hours of developers having religious discussions over checking for null before calling free() or letting free() handle it.

Reply:
We have a number of places when we do it this way, and the other way. Should we go over the whole things and make it uniform?